### PR TITLE
Temp workaround to longhorn mount in pid 1

### PR DESCRIPTION
**Pull request by andrewd-zededa**

### CODE REVIEW

**Comment by andrewd-zededa** [comment]
Addressed eve reboots and pod updates, now checks if pod contains change

File: pkg/kube/cluster-init.sh [hunk 1]
```diff
@@ -417,7 +435,7 @@ if [ ! -f /var/lib/all_components_initialized ]; then
                 apply_longhorn_disk_config $HOSTNAME
                 #kubectl apply -f  https://raw.githubusercontent.com/longhorn/longhorn/${LONGHORN_VERSION}/deploy/longhorn.yaml
                 # Switch back to above once all the longhorn services use the updated go iscsi tools
-                kubectl apply -f /etc/longhorn-config.yaml
+                kubectl apply -f /etc/longhorn-config.yaml
```
Comment (naiming-zededa) [inline pkg/kube/cluster-init.sh:438]: remove the space there.

File: pkg/kube/cluster-init.sh [hunk 2]
```diff
@@ -466,6 +484,8 @@ else
             else
                 logmsg "k3s is down and restart count exceeded."
             fi
+        else
+          check_overwrite_nsmounter
```
Comment (naiming-zededa) [inline pkg/kube/cluster-init.sh:488]: i don't think this call should be here.
it should be in 2 places, first on is inside the if [ ! -f /var/lib/longhorn_initialized ]; then ... fi block for the first time the longhorn is downloaded and started; the 2nd place should be inside the if [ $? -eq 1 ]; then ... fi block, and maybe before the "# setup debug ....", so when the device restarts, we also run this, but only once. Unless I misunderstood the function here.

Comment (andrewd-zededa) [reply pkg/kube/cluster-init.sh:488]: In this spot it runs after longhorn is installed and on restarts

**Review by naiming-zededa — APPROVED** [review]
LGTM

File: pkg/kube/cluster-init.sh (hunk 1 above)
Comment (andrewd-zededa) [reply pkg/kube/cluster-init.sh]: Should be fixed now

**Pull request merged**